packages/core/src/Parser.ts

```diff
@@ -130,70 +138,32 @@ export class Parser {
      * @returns The next parsing context
      */
     _parseNode(currentContext: ParsingContext): ParsingContext {
-        let context;
-        switch (currentContext.node.nodeType) {
-            case Node.ELEMENT_NODE: {
-                context = this._parseElementNode(currentContext);
-                break;
-            }
-            case Node.TEXT_NODE: {
-                context = this._parseTextNode(currentContext);
-                break;
-            }
-            case Node.DOCUMENT_NODE:
-            case Node.DOCUMENT_FRAGMENT_NODE: {
-                // These nodes have no effect in the context of parsing, but the
-                // parsing itself will continue with their children.
-                context = currentContext;
-                break;
-            }
-            case Node.CDATA_SECTION_NODE:
-            case Node.PROCESSING_INSTRUCTION_NODE:
-            case Node.COMMENT_NODE:
-            case Node.DOCUMENT_TYPE_NODE:
-            default: {
-                throw `Unsupported node type: ${currentContext.node.nodeType}.`;
-            }
-        }
-        return this._nextParsingContext(context);
-    }
-    /**
-     * Parse the given DOM Element into VNode(s).
-     *
-     * @param node to parse
-     * @param [format] to apply to the parsed node (default: none)
-     * @returns the parsed VNode(s)
-     */
-    _parseElementNode(currentContext: ParsingContext): ParsingContext {
         const context = { ...currentContext };
-
         const node = context.node;
-        const parsedNode = this.parseNode(node)[0] as VNode;
-        if (Format.tags.includes(context.node.nodeName)) {
-            // Format nodes (e.g. B, I, U) are parsed differently than regular
-            // elements since they are not represented by a proper VNode in our
-            // internal representation but by the format of its children.
-            // For the parsing, encountering a format node generates a new format
-            // context which inherits from the previous one.
-            const format = context.format.length ? context.format[0] : {};
-            context.format.unshift({
-                bold: format.bold || node.nodeName === 'B',
-                italic: format.italic || node.nodeName === 'I',
-                underline: format.underline || node.nodeName === 'U',
+        const parseResult = this.parseNode(node);
+        if (Array.isArray(parseResult)) {
```

Comment:
Part of the problem of the double return type of `parseNode`. This actually checks if it returns `VNodes` instead of attributes.